unordered_associative_task_2/student/src/impl.cpp

```diff
@@ -44,6 +69,19 @@ filtering_map groupDuplicates (const std::vector<std::string>& dataSource, Filte
 std::vector<std::vector<std::string> > findDuplicates(const std::string &rootPath)
 {
     // filter by size
-    // filter by content
+    filtering_map by_size = groupDuplicates(listFiles(rootPath), true);
+
     // flatten
+    std::vector<std::string> duplicated_file_list = flattenGrouped(by_size);
+
+    // filter by content
+    filtering_map by_hash = groupDuplicates(duplicated_file_list, false);
+    
+    std::vector<std::vector<std::string>> duplicated;
+    for( auto group : by_hash){
```

Comment:
by default, auto will be resolved as a plain type name (not a reference!), so add const auto&